tests/python/gaia-ui-tests/gaiatest/apps/clock/app.py

```diff
@@ -23,13 +19,21 @@ def launch(self):
         Wait(self.marionette).until(expected.element_displayed(
             Wait(self.marionette).until(expected.element_present(
                 *self._visible_clock_locator))))
-        Wait(self.marionette).until(expected.element_displayed(
-            Wait(self.marionette).until(expected.element_present(
-                *self._alarm_create_new_locator))))
 
-    @property
-    def alarms(self):
-        return [self.Alarm(self.marionette, alarm) for alarm in self.marionette.find_elements(*self._all_alarms_locator)]
+    def switch_view(self, view_name):
+        Wait(self.marionette).until(
```

Comment:
I think you can just as well use find_element here, see info: http://mxr.mozilla.org/mozilla-central/source/testing/marionette/driver/marionette_driver/marionette.py#1574
"
If no element is immediately found, the attempt to locate an element will be repeated for up to the amount of time set by set_search_timeout()
"